contracts/espace/VeShui.sol

```diff
@@ -0,0 +1,341 @@
+            settleUser(todayUnlockUsers[i]);
+        }
+        delete dayUnlockUsers[day];
+
+        lastSettleDay = day;
+        aprOnLastSeetleDay = (profit * 1e18 * 365) / totalVeShui;
+    }
+
+    function settleUser(address user) private {
+        userInfos[user].reward +=
+            (userInfos[user].totalVeShui * (accRewardPerVeShui - userInfos[user].userRewardPerTokenPaid)) /
+            1e18;
+        userInfos[user].userRewardPerTokenPaid = accRewardPerVeShui;
+    }
+
+    function withdraw() external {
+        uint today = date(block.timestamp);
+        uint256[] memory lockIndexes = userInfos[msg.sender].locks.values();
+
+        for (uint256 i = 0; i < lockIndexes.length; i++) {
+            uint256 lockIndex = lockIndexes[i];
+            if (locks[lockIndex].unlockDay <= today && !locks[lockIndex].isHandledUnlock) {
+                userInfos[msg.sender].unlockedShui += locks[lockIndex].amount;
+                totalLocked -= locks[lockIndex].amount;
+                locks[lockIndex].isHandledUnlock = true;
+            }
+        }
+
+        require(userInfos[msg.sender].unlockedShui > 0, "No unlocked shui");
+
+        uint256 userUnlockedShui = userInfos[msg.sender].unlockedShui;
+        userInfos[msg.sender].unlockedShui = 0;
+        shuiToken.transfer(msg.sender, userUnlockedShui);
+
+        emit Withdrawed(msg.sender, userUnlockedShui);
+    }
+
+    function claim() external {
+        settleUser(msg.sender);
+        require(userInfos[msg.sender].reward > 0, "No reward");
+
+        uint256 reward = userInfos[msg.sender].reward;
+        userInfos[msg.sender].reward = 0;
+        payable(msg.sender).transfer(reward);
+
+        emit Claimed(msg.sender, reward);
+    }
+
+    function getUserInfo(address _user) public view returns (Lock[] memory, uint, uint, uint) {
+        uint256[] memory lockIndexes = userInfos[_user].locks.values();
+        Lock[] memory userLocks = new Lock[](lockIndexes.length);
+        for (uint256 i = 0; i < lockIndexes.length; i++) {
+            userLocks[i] = locks[lockIndexes[i]];
+        }
+
+        uint reward = userInfos[_user].reward;
+        uint unlockedShui = userInfos[_user].unlockedShui;
+        uint _totalVeShui = userInfos[_user].totalVeShui;
+
+        uint today = date(block.timestamp);
+        for (uint256 i = 0; i < lockIndexes.length; i++) {
+            uint256 lockIndex = lockIndexes[i];
+
+            if (locks[lockIndex].lockDay <= today && !locks[lockIndex].isHandledLock) {
+                _totalVeShui += locks[lockIndex].veShuiAmount;
+            }
+
+            if (locks[lockIndex].unlockDay <= today && !locks[lockIndex].isHandledUnlock) {
+                unlockedShui += locks[lockIndex].amount;
+                _totalVeShui -= locks[lockIndex].veShuiAmount;
+            }
+        }
+
+        reward += (_totalVeShui * (accRewardPerVeShui - userInfos[_user].userRewardPerTokenPaid)) / 1e18;
```

Comment:
The calculated reward is more than he should get, one case:
day 1 user totalVeShui 100, the reward difference is 1, he should get 100,
day 2 user totalVeShui 200, the reward difference is another 1, he should get  200, 
total is 300.

but the reward calculated by formula above is 
200 * 2 = 400